File: python/leetcode165.py
class Solution(object):
    def compareVersion(self, version1, version2):
        """
        :type version1: str
        :type version2: str
        :rtype: int
        """
        v1 = version1.split('.')
        v2 = version2.split('.')
        i,j = 0,0
        while i < len(v1) and j < len(v2):
            if int(v1[i]) > int(v2[i]):
                return 1
            elif int(v1[i]) < int(v2[i]):
                return -1
            i += 1
            j += 1
        if i < len(v1) and int(''.join(v1[i:]))>0:
            return 1
        elif j < len(v2) and int(''.join(v2[j:]))>0:
            return -1
        else:
            return 0

File: leetcodec++/leetcode165.cpp
//
// Created by 黄敏辉 on 2019-08-25.
//

class Solution {
public:
    int compareVersion(string version1, string version2) {
        int l1 = version1.size();
        int l2 = version2.size();
        int v1 = 0, v2 = 0, i = 0 ,j = 0, n = 0, m = 0;

        while (i < l1 || j < l2){
            while (i < l1 && version1[i] != '.') {
                v1 = v1 * 10 + (version1[i] - '0');
                i++;
            }

            while (j < l2 && version2[j] != '.') {
                v2 = v2 * 10 + (version2[j] - '0');
                j++;
            }

            if (v1 > v2)
                return 1;
            if (v1 < v2)
                return -1;

            v1 = v2 = 0;
            i++;
            j++;

        }
        return 0;

    }
};